src/magic_folder/config.py

```diff
@@ -862,6 +855,21 @@ def store_local_snapshot(self, cursor, snapshot):
 
         :param LocalSnapshot snapshot: The snapshot to store.
         """
+        # Ensure that the local parent snapshots are already in the database.
+        for parent in snapshot.parents_local:
+            cursor.execute(
+                """
+                SELECT
+                    count(*) from [local_snapshots]
+                WHERE
+                    identifier= ?
+                """,
+                (unicode(parent.identifier),),
+            )
```

Comment:
I suppose that if we wanted this to be one query instead of N we could do dynamic SQL query string generation to make the `IN ...` expression to check all the parents at once.  But since N = 1 right now I guess there's no point.